### DISCUSSION TITLE
GEM port for ARM LINUX

### DISCUSSION THREAD

**Discussion by MikePooh** [Spirik/GEM#54] [General]
There is a port for u8g2 for [ARM LINUX](https://github.com/olikraus/u8g2/tree/master/sys/arm-linux). Is there any way to compile GEM for such type of devices?

**Comment by Spirik** [comment]
Hi, Mike!

I am not really sure, to be honest. I believe it should be possible - GEM doesn't heavily rely on Arduino-specific libraries or patterns (aside from PROGMEM directives and probably float implementation). But I can't easily tell how much of it would have to be rewritten to be successfully ported to ARM Linux system. Although that's really interesting perspective. I may look into it in more details when I have a chance.

Thank you for bringing it up!

**Reply by MikePooh** [reply]
It would be great!

**Comment by MikePooh** [comment]
Any news? )

**Reply by Spirik** [reply]
Unfortunately no, not at the time at least. Just to be clear: I don't think it is a matter of the nearest future, but something I'd like to explore a little bit further down the road. More pressing matter right now is reports of compatibility issues with some of the [ESP32](https://github.com/Spirik/GEM/issues/55) and [SAMD](https://github.com/Spirik/GEM/issues/60) boards.
